File: Rich_Game/src/com/tw/commands/UpgradeEstate.java
package com.tw.commands;

import com.tw.map.Estate;
import com.tw.player.Player;

/**
 * Created by pzzheng on 11/15/16.
 */
public class UpgradeEstate implements Responsive {
    private Estate estate;

    public UpgradeEstate(Estate estate) {
        this.estate = estate;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        UpgradeEstate that = (UpgradeEstate) o;

        return estate != null ? estate.equals(that.estate) : that.estate == null;

    }

    @Override
    public int hashCode() {
        return estate != null ? estate.hashCode() : 0;
    }

    @Override
    public Player.Status respond(Player player, ResponseType responseType) {
        player.decreaseFunds(estate.getEmptyPrice());
        estate.upgrade();
        return player.endTurn();
    }
}

File: ruby/rich_game_ruby/src/commands/upgrade_estate.rb
require_relative 'command'
require_relative 'command_factory'
class UpgradeEstate < Command
  def initialize(estate)
    @estate = estate
  end

  def respond(response, player)
    if response == CommandFactory::Yes
      player.asset.upgradeEstate(@estate)
      @estate.upgrade
    end
    player.endTurn
  end
  end